src/storage/src/manifest/storage.rs

```diff
@@ -321,6 +321,60 @@ impl ManifestLogStorage for ManifestObjectStore {
         Ok(ret)
     }
 
+    async fn delete_all(&self) -> Result<()> {
+        let entries: Vec<Entry> = self.get_paths(Some).await?;
+
+        // Before invoking `delete_all()`, it is crucial to ensure that there exists at least one delta file that contains a `RemoveAction`.
+        let largest_id = entries
+            .iter()
+            .filter_map(|e| {
+                let name = e.name();
+                if is_delta_file(name) {
+                    return Some(file_version(name));
+                }
+                None
+            })
+            .max()
+            .unwrap();
```

Comment:
I wonder if this unwrap is safe.

Reply:
Yes, we can ensure that there is at least one delta file that contains a `RegionRemove` action. but as you mentioned,  it would be better to directly pass the file versions with delete operations to this.